cpp/unit/ability.cpp

```diff
@@ -81,9 +88,23 @@ GarrisonAbility::GarrisonAbility(Sound *s)
 }
 
 bool GarrisonAbility::can_invoke(Unit &to_modify, const Command &cmd) {
-	return to_modify.location &&
-	       cmd.has_unit() &&
-	       cmd.unit()->has_attribute(attr_type::garrison);
+	if (cmd.has_unit()) {
+		Unit &target = *cmd.unit();
+
+		// make sure buildings are completed
+		if (target.has_attribute(attr_type::building)) {
+			auto &build_attr = target.get_attribute<attr_type::building>();
```

Comment:
Why doesn't `has_attribute` use the same template-parameter syntax as `get_attribute`?